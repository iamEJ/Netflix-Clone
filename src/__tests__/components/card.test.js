import React from "react";
import { render, fireEvent, getByAltText } from "@testing-library/react";
import Card from "../../components/card";
import Player from "../../components/player";

const category = "series";
const slideRows = [
  {
    title: "Comedy",
    data: [
      {
        id: "bb25cb15-1509-4f19-930a-ad3323074118",
        title: "The Office",
        description:
          "A motley group of office workers go through hilarious misadventures at the Scranton, Pennsylvania, branch of the Dunder Mifflin Paper Company.",
        genre: "comedies",
        maturity: "15",
        slug: "the-office",
        docId: "ZNgJ0KqUxbq6U9XNIGIb",
      },
    ],
  },
  {
    title: "Children",
    data: [
      {
        id: "9c5e35ca-6707-4bc5-b88e-8542c389d87b",
        title: "Peppa Pig",
        description:
          "Peppa, an outgoing preschool pig, participates in many energetic activities. She learns something new every day and has a lot of fun with her family and friends.",
        genre: "children",
        maturity: "0",
        slug: "peppa-pig",
        docId: "P7H1WR6YEnTE91wH2bU1",
      },
    ],
  },
];

describe("<Card />", () => {
  it("renders <Card /> with populated data", () => {
    const { container, getByText } = render(
      <Card.Group>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item) => (
                <Card.Item key={item.docId} item={item}>
                  <Card.Image src={item.image} />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src="/videos/bunny.mp4" />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );
    expect(getByText("Comedy")).toBeTruthy();
    expect(getByText("The Office")).toBeTruthy();
    expect(
      getByText(
        "A motley group of office workers go through hilarious misadventures at the Scranton, Pennsylvania, branch of the Dunder Mifflin Paper Company."
      )
    ).toBeTruthy();

    expect(getByText("Children")).toBeTruthy();
    expect(getByText("Peppa Pig")).toBeTruthy();
    expect(
      getByText(
        "Peppa, an outgoing preschool pig, participates in many energetic activities. She learns something new every day and has a lot of fun with her family and friends."
      )
    ).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders <Card /> and toggles card feature", () => {
    const { container, queryByText, getByTestId, getByAltText } = render(
      <Card.Group>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item) => (
                <Card.Item
                  key={item.docId}
                  item={item}
                  data-testid={`${item.slug}-item-feature`}
                >
                  <Card.Image src={item.image} />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src="/videos/bunny.mp4" />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );
    expect(queryByText("15")).toBeFalsy();
    fireEvent.click(getByTestId("the-office-item-feature"));
    expect(queryByText("15")).toBeTruthy();

    expect(queryByText("PG")).toBeFalsy();
    fireEvent.click(getByTestId("peppa-pig-item-feature"));
    expect(queryByText("PG")).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
