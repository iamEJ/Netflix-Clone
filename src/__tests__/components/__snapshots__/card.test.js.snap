// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders <Card /> and toggles card feature 1`] = `
<div
  class="card__Group-sc-29grb9-2 jXGfoc"
>
  <div
    class="card__Container-sc-29grb9-1 kezBSj"
  >
    <p
      class="card__Title-sc-29grb9-0 jwinrX"
    >
      Comedy
    </p>
    <div
      class="card__Entities-sc-29grb9-12 kkpZCO"
    >
      <div
        class="card__Item-sc-29grb9-13 kjzviD"
        data-testid="the-office-item-feature"
      >
        <img
          class="card__Image-sc-29grb9-14 bERlgv"
        />
        <div
          class="card__Meta-sc-29grb9-11 bQUeoW"
        >
          <p
            class="card__SubTitle-sc-29grb9-3 knpugP"
          >
            The Office
          </p>
          <p
            class="card__Text-sc-29grb9-4 byIjVO"
          >
            A motley group of office workers go through hilarious misadventures at the Scranton, Pennsylvania, branch of the Dunder Mifflin Paper Company.
          </p>
        </div>
      </div>
    </div>
    <div
      class="card__Feature-sc-29grb9-6 dHzfGw"
    >
      <div
        class="card__Content-sc-29grb9-10 fMvTeq"
      >
        <p
          class="card__Title-sc-29grb9-0 card__FeatureTitle-sc-29grb9-7 jwinrX eSxjXq"
        >
          The Office
        </p>
        <p
          class="card__FeatureText-sc-29grb9-5 ifdRUv"
        >
          A motley group of office workers go through hilarious misadventures at the Scranton, Pennsylvania, branch of the Dunder Mifflin Paper Company.
        </p>
        <button
          class="card__FeatureClose-sc-29grb9-8 lorGTD"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
        <div
          class="card__Group-sc-29grb9-2 dWhOPk"
        >
          <div
            class="card__Maturity-sc-29grb9-9 gGJvMp"
          >
            15
          </div>
          <p
            class="card__FeatureText-sc-29grb9-5 ifdRUv"
          >
            Comedies
          </p>
        </div>
        <div
          class="player__Container-zumstd-0"
        >
          <button
            class="player__Button-zumstd-3 cfkfEQ"
          >
            Play
          </button>
        </div>
      </div>
    </div>
  </div>
  ;
  <div
    class="card__Container-sc-29grb9-1 kezBSj"
  >
    <p
      class="card__Title-sc-29grb9-0 jwinrX"
    >
      Children
    </p>
    <div
      class="card__Entities-sc-29grb9-12 kkpZCO"
    >
      <div
        class="card__Item-sc-29grb9-13 kjzviD"
        data-testid="peppa-pig-item-feature"
      >
        <img
          class="card__Image-sc-29grb9-14 bERlgv"
        />
        <div
          class="card__Meta-sc-29grb9-11 bQUeoW"
        >
          <p
            class="card__SubTitle-sc-29grb9-3 knpugP"
          >
            Peppa Pig
          </p>
          <p
            class="card__Text-sc-29grb9-4 byIjVO"
          >
            Peppa, an outgoing preschool pig, participates in many energetic activities. She learns something new every day and has a lot of fun with her family and friends.
          </p>
        </div>
      </div>
    </div>
    <div
      class="card__Feature-sc-29grb9-6 dHzfGw"
    >
      <div
        class="card__Content-sc-29grb9-10 fMvTeq"
      >
        <p
          class="card__Title-sc-29grb9-0 card__FeatureTitle-sc-29grb9-7 jwinrX eSxjXq"
        >
          Peppa Pig
        </p>
        <p
          class="card__FeatureText-sc-29grb9-5 ifdRUv"
        >
          Peppa, an outgoing preschool pig, participates in many energetic activities. She learns something new every day and has a lot of fun with her family and friends.
        </p>
        <button
          class="card__FeatureClose-sc-29grb9-8 lorGTD"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
        <div
          class="card__Group-sc-29grb9-2 dWhOPk"
        >
          <div
            class="card__Maturity-sc-29grb9-9 jwYEld"
          >
            PG
          </div>
          <p
            class="card__FeatureText-sc-29grb9-5 ifdRUv"
          >
            Children
          </p>
        </div>
        <div
          class="player__Container-zumstd-0"
        >
          <button
            class="player__Button-zumstd-3 cfkfEQ"
          >
            Play
          </button>
        </div>
      </div>
    </div>
  </div>
  ;
</div>
`;

exports[`<Card /> renders <Card /> with populated data 1`] = `
<div
  class="card__Group-sc-29grb9-2 jXGfoc"
>
  <div
    class="card__Container-sc-29grb9-1 kezBSj"
  >
    <p
      class="card__Title-sc-29grb9-0 jwinrX"
    >
      Comedy
    </p>
    <div
      class="card__Entities-sc-29grb9-12 kkpZCO"
    >
      <div
        class="card__Item-sc-29grb9-13 kjzviD"
      >
        <img
          class="card__Image-sc-29grb9-14 bERlgv"
        />
        <div
          class="card__Meta-sc-29grb9-11 bQUeoW"
        >
          <p
            class="card__SubTitle-sc-29grb9-3 knpugP"
          >
            The Office
          </p>
          <p
            class="card__Text-sc-29grb9-4 byIjVO"
          >
            A motley group of office workers go through hilarious misadventures at the Scranton, Pennsylvania, branch of the Dunder Mifflin Paper Company.
          </p>
        </div>
      </div>
    </div>
  </div>
  ;
  <div
    class="card__Container-sc-29grb9-1 kezBSj"
  >
    <p
      class="card__Title-sc-29grb9-0 jwinrX"
    >
      Children
    </p>
    <div
      class="card__Entities-sc-29grb9-12 kkpZCO"
    >
      <div
        class="card__Item-sc-29grb9-13 kjzviD"
      >
        <img
          class="card__Image-sc-29grb9-14 bERlgv"
        />
        <div
          class="card__Meta-sc-29grb9-11 bQUeoW"
        >
          <p
            class="card__SubTitle-sc-29grb9-3 knpugP"
          >
            Peppa Pig
          </p>
          <p
            class="card__Text-sc-29grb9-4 byIjVO"
          >
            Peppa, an outgoing preschool pig, participates in many energetic activities. She learns something new every day and has a lot of fun with her family and friends.
          </p>
        </div>
      </div>
    </div>
  </div>
  ;
</div>
`;
